### ISSUE TITLE
[cdk8s-plus] Add an IAM PolicyStatement to ServiceAccount

### ISSUE THREAD

**Issue by van-vothanh** [awslabs/cdk8s#356]
With the `@aws-cdk/aws-eks` package, it is possible to add an IAM policy statement to a `ServiceAccount` through the this method https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-eks.ServiceAccount.html#add-wbr-to-wbr-principal-wbr-policystatement-span-class-api-icon-api-icon-experimental-title-this-api-element-is-experimental-it-may-change-without-notice-span

How do we achieve the same using `cdk8-plus`'s `ServiceAccount` construct?

Thank you

**Comment by iliapolo** [comment]
Hi @van-vothanh the `ServiceAccount` in `cdk8s-plus` is not natively integrated with AWS IAM roles. 
You can create an `iam.Role` using the `@aws-cdk/aws-iam` package, add the desired policies and then annotate the service account with the role arn.

Essentially it would mean re-implementing the logic in the [`@aws-cdk/aws-eks.ServiceAccount`](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-eks/lib/service-account.ts) construct.

**Comment by van-vothanh** [comment]
hi @iliapolo thanks for the clarification.

Do you have a code sample showing how we can annotate a service account with the role arn?

Cheers

**Comment by van-vothanh** [comment]
hi @iliapolo 

I managed to get everything working as expected, however the experience wasn't as smooth as it was with `cdk-eks` previously. In addition to the `iam.Role`, I also had to create a new `iam.OpenIdConnectProvider` in my `cdk` app, this is where things get interesting. The `iam.OpenIdConnectProviderProps` doesn't expose the ability to set the `audience` nor the `CA thumbprint`, in the end I had to manually go on to the `IAM` console to configure these.

As you probably are aware, the `cdk-eks` constructs take care of all that for me. I still think there is value in replicating this behaviour in the `cdk8s` or even `cdk8s+` constructs.

Thanks
